charts/ingress-nginx/templates/_helpers.tpl

```diff
@@ -146,3 +146,12 @@ Return the appropriate apiVersion for podSecurityPolicy.
 {{- print "extensions/v1beta1" -}}
 {{- end -}}
 {{- end -}}
+
+{{/*
+Check the ingress controller image tag
+*/}}
+{{- define "isControllerTagValid" -}}
+{{- if semverCompare ">=0.27.0" .Values.controller.image.tag -}}
+true
+{{- end -}}
```

Comment:
```suggestion
{{- if not (semverCompare ">=0.27.0" .Values.controller.image.tag) -}}
{{- fail "Controller container image tag should be 0.27.0 or higher" -}}
{{- end -}}
```

Reply:
I tried that but it doesn't work

```
{{- if not (semverCompare ">=0.27.0" .Values.controller.image.tag) -}}
{{- fail "Controller container image tag should be 0.27.0 or higher" -}}
{{- end -}}
```
```
helm template test2  $PWD/charts/ingress-nginx --set controller.image.tag=0.26.1 --debug
install.go:158: [debug] Original chart version: ""
install.go:175: [debug] CHART PATH: /home/aledbf/go/src/k8s.io/ingress-nginx/charts/ingress-nginx
```

Reply:
Using helm2 this works
```
Error: render error in "ingress-nginx/templates/controller-deployment.yaml": template: ingress-nginx/templates/controller-deployment.yaml:3:4: executing "ingress-nginx/templates/controller-deployment.yaml" at <fail "Controller container image tag should be 0.27.0 or higher">: error calling fail: Controller container image tag should be 0.27.0 or higher
```

Reply:
huh, I'm using helm 3.1.1 and I'm seeing the desired result.
in controller-deployment.yaml I have added below the if statement
```
{{- include  "isControllerTagValid" . -}}
```

running the same command as you:
```
╰─ helm template ./ingress-nginx --set controller.image.tag=0.26.1
Error: template: ingress-nginx/templates/controller-deployment.yaml:2:4: executing "ingress-nginx/templates/controller-deployment.yaml" at <include "isControllerTagValid" .>: error calling include: template: ingress-nginx/templates/_helpers.tpl:155:4: executing "isControllerTagValid" at <fail "Controller container image tag should be 0.27.0 or higher">: error calling fail: Controller container image tag should be 0.27.0 or higher
```

Reply:
You are right, I was using helm 3.0.1. After the update to 3.1.2 it works

Reply:
weird